mcp/server.py

```
@@ -0,0 +1,522 @@
+        args.append("--enable-write-endpoints")
+    if config_file:
+        cfg = _repo_path(config_file)
+        if not cfg.is_file():
+            return f"config file not found: {cfg}"
+        args.extend(["--config", str(cfg)])
+
+    _ensure_dev_log_dir()
+    log_path.parent.mkdir(parents=True, exist_ok=True)
+    logf = open(log_path, "a", encoding="utf-8")
+    try:
+        proc = subprocess.Popen(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)